### DISCUSSION TITLE
The date in the macro is 31-12-1999 and the time is 22.00

### DISCUSSION THREAD

**Discussion by issa-mokhtar** [WarmUpTill/SceneSwitcher#656] [General]
I updated OBS and also Aadvanced scene switcher and when entering the old settings all dates and hours appear in the macro with one date 12-31-1999 and 22.00

**Comment by WarmUpTill** [comment]
Which OBS version / plugin version were you using previously?
Are you able to share the settings file?

**Reply by issa-mokhtar** [reply]
I'm using OBS 27.2.4 with v1.17.4 of the Advanced Scene Switcher plug-in.
 Then I updated OBS 29 and updated the plugin to 1.19.2
 Unfortunately, after the update, I encountered this problem.

**Reply by WarmUpTill** [reply]
Did you manually import the setting via the "Import" option on the General tab from a file?
If so, can you share it?

If not, do you maybe still have a backup of your scene collection before updating?
If so, would you be willing to share it, so I can try to reproduce the problem?

**Reply by issa-mokhtar** [reply]
[إعدادات سنة 2023.txt](https://github.com/WarmUpTill/SceneSwitcher/files/10416872/2023.txt)

**Reply by issa-mokhtar** [reply]
Yes  I Did manually import the setting via the "Import" option on the General tab from a file.
I sent the settings file in the attachments

**Reply by WarmUpTill** [reply]
To be honest I have no idea why this is happening. :(

I can only imagine that the transition from Qt5 to Qt6 somehow influenced how date strings are represented in Arabic and thus the old date strings are no longer recognized by Qt6 and they all fall back to the date you mentioned.

Are modifications to the dates saved correctly or are they always reset to "31-12-1999 ..."?

I have attempted to fix at least some of the values by running a small script over the "00.00", "00.30", ... macros.
For the rest I was not sure what was intended.
I have attached this file below - I hope that helps at least a bit.
[fixed.txt](https://github.com/WarmUpTill/SceneSwitcher/files/10417577/fixed.txt)

Just for reference here is the python script I used to fix at least some of the values:
``` 
import json
import re


def handle_date_cond(condition, time):
    time = time.replace(".", ":")
    if time.startswith("24"):
        time = "00" + time[2:]
    condition["dateTime"] = "Sat Jan 14 {}:00 2023".format(time)


def is_time_string(name):
    pattern = re.compile("^\d\d\.\d\d$")
    if pattern.match(name):
        return True
    return False


def main():
    json_data = ""
    with open("input.txt", "r", encoding="utf-8") as json_file:
        lines = json_file.readlines()
        for line in lines:
            json_data += line

    data = json.loads(json_data)

    for macro in data["macros"]:
        macro_name = macro["name"]
        if not is_time_string(macro_name):
            continue
        for condition in macro["conditions"]:
            if condition["id"] == "date":
                handle_date_cond(condition, macro_name)

    with open("out.txt", "w") as outfile:
        outfile.write(json.dumps(data, indent=4))


if __name__ == "__main__":
    main()
```

**Reply by issa-mokhtar** [reply]
Thank you very much, my dear friend. I reorganized the settings again on the new version and wrote in English letters, not Arabic, and so far the situation is excellent.

**Reply by issa-mokhtar** [reply]
Is it possible to add a "full screen projector preview" option to a macro so that it runs at a specific time?

**Reply by WarmUpTill** [reply]
I am not sure I understand what you mean by that.
Can you give an example what you would want to achieve with this?

**Reply by issa-mokhtar** [reply]
I mean that the plugin will do a "fullscreen projector preview" at a certain time or after a certain condition is met. I am using OBS in a visual channel in my city, the plugin displays the scenes at specific times that I set, the image is displayed in the channel by HDMI cable except for the computer monitor cable, and every day it goes to a different scene collection, but when I change to scene collection  new the program locks the display of the image to the hdmi of the channel, and i need to display it again, i want the plugin to do this automatically each time it changes to a new scene collection at a specified time. Pictures attached

**Reply by WarmUpTill** [reply]
I see - that should be doable.
I am just not sure when I will get around to it.

**Reply by issa-mokhtar** [reply]
Thank you so much.  I'm waiting for you, I hope you can find a way.  Can we communicate via WhatsApp or Telegram?

**Comment by issa-mokhtar** [comment]
![Screenshot_20230115_183320_AnyDesk](https://user-images.githubusercontent.com/122641239/212553641-9423fd3a-9875-4ef5-9ab5-2d3fcc6f9afa.jpg)
![Screenshot_20230115_181738_AnyDesk](https://user-images.githubusercontent.com/122641239/212553642-92b95668-1365-4811-b211-c1a6599bbbf4.jpg)

**Comment by issa-mokhtar** [comment]
Can only one macro be exported and then re-imported again? I do not mean all settings.

**Reply by WarmUpTill** [reply]
No, that is currently not possible.

**Reply by issa-mokhtar** [reply]
We hope it will be present in the next update.  Thank you so much
